lib/view/weather_view/component/weather_forecast.dart

```diff
@@ -15,11 +15,11 @@ class WeatherForecast extends StatelessWidget {
       children: [
         AspectRatio(
           aspectRatio: 1,
-          child: weatherCondition == null
+          child: weatherInfo == null
               ? const Placeholder()
               : SvgPicture.asset(
-                  'images/${weatherCondition!.name}.svg',
-                  semanticsLabel: '${weatherCondition!.name} image',
+                  'images/${weatherInfo?.weatherCondition.name}.svg',
+                  semanticsLabel: '${weatherInfo?.weatherCondition.name} image',
```

Comment:
[nits]
`!` のほうが適してそうです。

```suggestion
                  'images/${weatherInfo!.weatherCondition.name}.svg',
                  semanticsLabel: '${weatherInfo!.weatherCondition.name} image',
```

このように `!` を使用したくない場合は、ローカル変数に定義しなおしたり、 `null` のときとそうでないときそれぞれの `Widget` をプライベートクラスで作成するなどするといいかなと思います〜

Reply:
ちなみにローカル変数にすると、値変更の可能性がなくなりnon-nullable 型へ昇格できるため、 `!` を書く必要がなくなります。

なぜ、クラスのメンバーでは `final` で定義していてもダメなのかは↓を参考にしていただければと思います。

- https://dart.dev/tools/non-promotion-reasons
- https://qiita.com/kabochapo/items/fbca2b2972546eeadcfa

Reply:
> ちなみにローカル変数にすると、値変更の可能性がなくなりnon-nullable 型へ昇格できるため、 ! を書く必要がなくなります。

こちら知らなかったです... 勉強になりました！ありがとうございます🙇‍♂️

`null` の時とそうでない時それぞれの `Widget` をプライベートクラスで作成するとコード量の増大 & 重複コードが多くなってしまうと思ったので、ローカル変数に定義して`?`or`!`を削除する方針で実装しました！
32b4da8478e9433999e04e29a1a0226f4ad931cd